### PR TITLE
Remove the LLVM/bindgen/cmake dependencies from wasmtime-runtime

### PR DESCRIPTION
**Pull request by alexcrichton**
This involved moving around where setjmp/longjmp happen as well as tweaking some various interfaces, but it's intended that this is largely functionality neutral otherwise. Some more details can be found in the commits as well.

### CODE REVIEW

File: wasmtime-runtime/signalhandlers/Trampolines.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include <setjmp.h>
+
+#include "SignalHandlers.hpp"
+
+extern "C"
+int WasmtimeCallTrampoline(void *vmctx, void (*body)(void*, void*), void *args) {
+  jmp_buf buf;
+  void *prev;
```
Comment (sunfishcode) [inline wasmtime-runtime/signalhandlers/Trampolines.cpp]: This should be volatile, `void *volatile prev`, to ensure that we see the correct value after the longjmp.

File: wasmtime-runtime/signalhandlers/Trampolines.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  prev = EnterScope(&buf);
+  body(vmctx, args);
+  LeaveScope(prev);
+  return 1;
+}
+
+extern "C"
+int WasmtimeCall(void *vmctx, void (*body)(void*)) {
+  jmp_buf buf;
+  void *prev;
```
Comment (sunfishcode) [inline wasmtime-runtime/signalhandlers/Trampolines.cpp]: Likewise.

**Comment by alexcrichton** [comment]
Ok, should be updated! Good point about `volatile`, I've only barely used setjmp/longjmp myself in actual code...

**Comment by tschneidereit** [comment]
I'll merge this, given that @sunfishcode approved it with the requested changes made.

**Pull request merged**